lib/bloc/podcast_clip/podcast_clip_bloc.dart

```diff
@@ -0,0 +1,206 @@
+
+    String episodeUrl = clipDetails.episodeDetails.contentUrl;
+    String audioClipPath = await initClipDirectory(isVideoClip: false);
+    audioClipPath = audioClipPath + '/clip.mp3';
+
+    //Getting the raw audio clip
+    String clippedAudioCommand =
+        '-i "$episodeUrl" -ss ${clipDetails.startTimeStamp.inSeconds} -to ${clipDetails.endTimeStamp.inSeconds}  -acodec copy $audioClipPath';
+
+    FFmpegSession ffpegSessionDetails =
+        await FFmpegKit.execute(clippedAudioCommand);
+
+    final returnCode = await ffpegSessionDetails.getReturnCode();
+
+    if (ReturnCode.isSuccess(returnCode)) {
+      return audioClipPath;
+    } else {
+      String failedStackTrace = await ffpegSessionDetails.getFailStackTrace();
+
+      throw failedStackTrace;
+    }
+  }
+
+  getVideoClipPath({String audioClipPath, String episodeImagepath}) async {
+    String videoClipPath = await initClipDirectory(isVideoClip: true);
+    videoClipPath = videoClipPath + '/VideoClip.mp4';
+
+    String clippedVideoCommand =
+        '-i  $audioClipPath -i $episodeImagepath -filter_complex "[0:a]showwaves=colors=0xffffff@0.9:mode=p2p,format=yuva420p[v];[1:v]scale=2120:2560[bg];[bg][v]overlay=(main_w-overlay_w)/2:H*0.8[outv]" -map "[outv]" -map 0:a -c:v libx264 -c:a copy $videoClipPath';
+
+    FFmpegSession ffpegSessionDetails =
+        await FFmpegKit.execute(clippedVideoCommand);
+
+    final returnCode = await ffpegSessionDetails.getReturnCode();
+    if (ReturnCode.isSuccess(returnCode)) {
+      return videoClipPath;
+    } else {
+      String failedStackTrace = await ffpegSessionDetails.getFailStackTrace();
+
+      throw failedStackTrace;
+    }
+  }
+
+  bool isEpisodeClipable() {
+    var clipDetails = getCurrentPodcastClipDetails();
+
+    // Check if there is time to create a clip of minimun 10 secs
+    if ((clipDetails.endTimeStamp + Duration(seconds: _initialSeconds)) >
+        clipDetails.episodeLength) {
+      return false;
+    }
+    return true;
+  }
+
+  Future<String> clipEpisode({Uint8List clipImage}) async {
+    final directory = await getTemporaryDirectory();
+    final imageFile = await File('${directory.path}/image.png').create();
+    await imageFile.writeAsBytes(clipImage);
+
+    String podcastImageWidgetPath = imageFile.path;
+
+    var clipDetails = getCurrentPodcastClipDetails();
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        podcastClipState: PodcastClipState.FETCHING_AUDIO_CLIP));
+    if (clipDetails == null) {
+      return null;
+    }
+    String audioClipPath = await _getAudioClipPath(clipDetails: clipDetails);
+
+    if (audioClipPath == null) {
+      return null;
+    }
+
+    String videoClipPath = await getVideoClipPath(
+        audioClipPath: audioClipPath, episodeImagepath: podcastImageWidgetPath);
+    _clipDetailsBehaviourSubject
+        .add(clipDetails.copy(podcastClipState: PodcastClipState.IDLE));
+
+    if (videoClipPath == null) {
```

Comment:
Same here, we don't need this condition. In general it is better to count on exception thrown instead of returning null as exceptions contains more information.